### ISSUE TITLE
Refactor fixture based tests to use `broccoli-test-helper`'s `createTempDir`

### ISSUE THREAD

**Issue by rwjblue** [ember-template-lint/ember-template-lint#854]
This will allow the tests to be reasoned about much easier. 

For a concrete example, I'd like for this test:

https://github.com/ember-template-lint/ember-template-lint/blob/996456138ed78926b40cf7bba109f281ebbf1e45/test/unit/get-config-test.js#L25-L34

To be rewritten like:

```js
  it('if config is provided, it is returned', function() {
    let expected = {
      rules: {
        foo: 'bar',
        baz: 'derp',
      },
    };

    project.write({
      '.template-lintrc.js': `module.exports = ${JSON.stringify(expected)};`,
    });

    process.chdir(project.path());
   
    let actual = getConfig({});
    expect(actual.rules).to.deep.equal(expected.rules);
  });
```

**Comment by locks** [comment]
This has been implemented by linked https://github.com/ember-template-lint/ember-template-lint/pull/876 PR.
